posthog/api/property_definition.py

```diff
@@ -91,10 +93,12 @@ def with_event_property_filter(
         is_filtering_by_event_names = event_names and len(event_names) > 0
 
         if is_filtering_by_event_names or is_event_property is not None:
-            event_property_field = "case when posthog_eventproperty.id is null then false else true end"
+            event_property_field = (
+                f"case when {self.posthog_eventproperty_table_join_alias}.id is null then false else true end"
+            )
 
         if is_filtering_by_event_names:
-            event_name_join_filter += " AND posthog_eventproperty.event in %(event_names)s"
+            event_name_join_filter += " AND event in %(event_names)s"
```

Comment:
This is probably a bug waiting to happen...

The filter is being used in the derived table definition and in matching against it

So, it works because there is no clash on `event`

This fixes the bug... and then I'll follow-up with some more care on if I have to use this in two places